infrastructure/lib/dns-stack.ts

```diff
@@ -0,0 +1,67 @@
+import * as cdk from '@aws-cdk/core';
+import * as apigw from '@aws-cdk/aws-apigateway';
+import { Function } from '@aws-cdk/aws-lambda';
+import { ARecord, HostedZone, RecordTarget } from '@aws-cdk/aws-route53';
+import * as route53Targets from '@aws-cdk/aws-route53-targets';
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import { SPADeploy } from 'cdk-spa-deploy';
+
+export interface DNSStackProps extends cdk.StackProps {
+    lambda: Function;
+}
+
+
+export class DNSStack extends cdk.Stack {
+    constructor(scope: cdk.Construct, id: string, props: DNSStackProps) {
+        super(scope, id, props);
+
+        const zone = HostedZone.fromLookup(this, `${id}-hz`, {
+            domainName: 'hobbycribbage.com',
+        });
+        // TODO it's safer to use `addResource` or `addMethod` or `addProxy` to point to this lambda
+        // than to have `proxy: true,`, because that's gonna just route every request to this lambda.
+        const restApi = new apigw.LambdaRestApi(this, 'cribbage-lambda-rest-api', {
+            restApiName: 'lambda-rest-api',
+            description: 'the bundled up golang server acting as a lambda',
+            handler: props.lambda,
+            proxy: true,
+            deployOptions: {
+                stageName: 'urlprefix',
```

Comment:
Is `'urlprefix'` a special stage name, or will this make our URL `http://something/urlprefix/`?